lib/data/entity.js

```diff
@@ -97,13 +101,13 @@ const extractEntity = (body, propertyNames, existingEntity) => {
   } else {
     entity = { system: Object.create(null), data: Object.create(null) };
     if (body.uuid && typeof body.uuid !== 'string')
-      throw Problem.user.invalidEntity({ reason: 'Value for [uuid] is not a string.' });
+      throw Problem.user.invalidDataTypeOfParameter({ value: 'uuid', expected: 'string' });
```

Comment:
It looks like `Problem.user.invalidDataTypeOfParameter` is expecting a value, not a field/property name. Should we pass `body.uuid` for `value` instead of `'uuid'`? Or maybe modify the Problem function to take either a value or a field name? Actually, it looks like elsewhere in the code is trying to pass in a field name, so that would probably be something useful to support:

https://github.com/getodk/central-backend/blob/d113ed2111d52411672a770a7a8b31b69bb13632/lib/http/preprocessors.js#L144-L148

Reply:
What should be the message? Right now the message is `Invalid input data type: expected '${value}' to be (${expected})`, I feel the sentence works for both `value` and `field`, however we should pick one for the sake of consistency.

Code snippet above is buggy. If you debug `should throw an error if a non-numeric offset is given` test, you'll see that `error.message` is "Invalid input data type: expected 'undefined' to be (integer)"

Reply:
It seems like most of the time, we want to pass a field name to it. The only time we pass it a value is when there's a Postgres error:

https://github.com/getodk/central-backend/blob/0661a7b8a122063808301f2c03ab09b76364eb8c/lib/util/db.js#L459-L464

I think that's the original use of the Problem. In that case, I don't think a field name is available, only a value (even though the value is named `field` in the code 😅). But everything since this original case has wanted to pass in a field. The field or value appears in the message, and it also appears in the Problem details object.

What do you think about having the Problem function accept either `field` or `value`? They seem like separate but valid cases. That'd be something like:

```js
invalidDataTypeOfParameter: problem(400.11, ({ field, value, expected }) => {
  const fieldOrValue = field ?? `'${value}'`;
  return `Invalid input data type: expected ${fieldOrValue} to be (${expected})`;
},
```

Maybe that's too complicated for a Problem function though? 🤔 I also think it'd be fine to use either `value` or `field` everywhere. If you decide to standardize on `value`, would you mind fixing the code snippet above? Or that could happen in a separate PR.